pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2404,11 +2404,19 @@ func buildListener(port api.ServicePort, annotations map[string]string, sslPorts
 	return listener, nil
 }
 
+func nodeNames(nodes []*api.Node) sets.String {
```

Comment:
This is copy-pasted 3 times in the different providers.  Maybe we can share it in pkg/cloudprovider/cloud.go?